genesis/process/shardGenesisBlockCreator.go

```diff
@@ -303,3 +319,71 @@ func createProcessorsForShard(arg ArgsGenesisBlockCreator) (*genesisProcessors,
 		rwdProcessor:  rewardsTxProcessor,
 	}, nil
 }
+
+func deployInitialSmartContracts(
+	processors *genesisProcessors,
+	arg ArgsGenesisBlockCreator,
+) error {
+	smartContracts, err := arg.SmartContractParser.InitialSmartContractsSplitOnOwnersShards(arg.ShardCoordinator)
+	if err != nil {
+		return err
+	}
+
+	currentShardSmartContracts := smartContracts[arg.ShardCoordinator.SelfId()]
+
+	for _, sc := range currentShardSmartContracts {
+		err = deployInitialSmartContract(processors, sc, arg)
+		if err != nil {
+			return fmt.Errorf("%w for owner %s and filename %s",
+				err, sc.GetOwner(), sc.GetFilename())
+		}
+	}
+
+	return nil
+}
+
+func getSCCodeAsHex(filename string) (string, error) {
+	code, err := ioutil.ReadFile(filepath.Clean(filename))
+	if err != nil {
+		return "", err
+	}
+
+	return hex.EncodeToString(code), nil
+}
+
+func deployInitialSmartContract(
+	processors *genesisProcessors,
+	sc genesis.InitialSmartContractHandler,
+	arg ArgsGenesisBlockCreator,
+) error {
+	code, err := getSCCodeAsHex(sc.GetFilename())
+	if err != nil {
+		return err
+	}
+
+	vmType := sc.GetVmType()
+	deployTxData := strings.Join([]string{code, vmType, codeMetadataHex}, "@")
+
+	tx := &transactionData.Transaction{
+		Nonce:     0,
```

Comment:
nonce according to owner nonce if he has more to deploy.

Reply:
Agree, this is a must as an address can deploy multiple contracts